### PR TITLE
MagicScroll and FullZoom ShortKeys

### PR DESCRIPTION
**Pull request by Sheepux**
Using spacebar, if the scroll down goes lower than the image, it will stop at the bottom of this image (if the scroll is not enougth to go lower than the image because of zoom then it will be a normal scroll). 
When you've reached the bottom of an image, pressing on spacebar again will scroll down to the top of the next image.

Also adding shorcuts (contributes to #1859 )
- FullImageWidth zoom * (zoom 100)
- DefaultImageWidth zoom / (zoom 75)

### CODE REVIEW

File: src/web/lib/hakuneko/frontend@classic-dark/pages.html [hunk 1]
```diff
@@ -516,6 +525,42 @@
...
             }
+
+            /**
+             * Dynamically change the scrolling to stop at the end of images or skip to the start of the next image
+             * Usefull for manga reading but ugly with webtoons
+             */
+            scrollMagic(defaultDistance) {
+                let images = this.$.container.getElementsByClassName('image');
+                // Find current image within view
+                let currentScrollImages = Array.prototype.filter.call(images, function(image){
```
Comment (27136773) [inline src/web/lib/hakuneko/frontend@classic-dark/pages.html]: May use spread operator `[...images].filter(image => { /* fancy stuff here */ })`

File: src/web/lib/hakuneko/frontend@classic-dark/pages.html [hunk 2]
```diff
@@ -516,6 +525,42 @@
                     this.$.container.scrollTop = previousOffset * this.$.container.scrollHeight / previousHeight;
                 //}.bind( this ), 0 );
             }
+
+            /**
+             * Dynamically change the scrolling to stop at the end of images or skip to the start of the next image
+             * Usefull for manga reading but ugly with webtoons
+             */
+            scrollMagic(defaultDistance) {
+                let images = this.$.container.getElementsByClassName('image');
```
Comment (27136773) [inline src/web/lib/hakuneko/frontend@classic-dark/pages.html]: May use `container.querySelectorAll('image')`

**Review by 27136773 — APPROVED** [review]
Looks good

File: src/web/lib/hakuneko/frontend@classic-dark/pages.html (hunk 1 above)
Comment (Sheepux) [reply src/web/lib/hakuneko/frontend@classic-dark/pages.html]: Oh, seems easier to read indeed. Never used spread op before (quite the old fashion one)

File: src/web/lib/hakuneko/frontend@classic-dark/pages.html (hunk 2 above)
Comment (Sheepux) [reply src/web/lib/hakuneko/frontend@classic-dark/pages.html]: Indeed, easier to expand if needed.

**Pull request merged**